File: 151002/Luschickij/fullProject/DC_lab/discussion/src/main/java/com/luschickij/discussion/dto/post/PostResponseTo.java
package com.luschickij.discussion.dto.post;

import com.fasterxml.jackson.annotation.JsonRootName;
import com.luschickij.discussion.model.IdentifiedEntity;
import lombok.*;

@EqualsAndHashCode(callSuper = true)
@Data
@Builder
@JsonRootName("post")
@AllArgsConstructor
@NoArgsConstructor
public class PostResponseTo extends IdentifiedEntity {

    private Long id;
    private Long newsId;
    private String content;
    private String country;
}

File: 151004/Danilov/publisher/src/main/kotlin/com/danilovfa/model/post/PostResponseTo.kt
package com.danilovfa.model.post

import kotlinx.serialization.Serializable

@Serializable
data class PostResponseTo(
    val id: Long,
    val storyId: Long,
    val content: String
)

fun Post.toResponse() =
    PostResponseTo(
        id = id,
        storyId = storyId,
        content = content
    )
